doc/cabal-package-description-file.rst

```diff
@@ -2671,8 +2671,42 @@ Source Repositories
 .. pkg-section:: source-repository
     :since: 1.6
 
-It is often useful to be able to specify a source revision control
-repository for a package. Cabal lets you specify this information in
+.. warning::
+
+    A ``source-repository`` is for a version of the source of its package
```

Comment:
I'm not a fan of the current version of this warning. My version:

Note that ``source-repository-package`` is not related to the ``source-repository`` field in package descriptions. [link the field]
The ``source-repository`` field specifies the source repository where upstream development of the package is happening. It is currently only used by ``cabal get`` via the same-named parameter if specified. [Link to the docs of the parameter] Whereas, the ``source-repository-package`` project stanza adds a local package to the project much like listing the package's source directory in the ``packages`` field, except that the source code is downloaded as needed instead of being taken from the local directory.

---

Also, the warning should go after the stanza has been introduced.

Reply:
> via the same-named parameter if specified

Sorry I don't quite follow @ulysses4ever. Do you mean `[PACKAGES]` by this?

```
$ cabal get --help
Download/Extract a package's source code (repository).

Usage: cabal get [PACKAGES]
...
```

Reply:
> Note that `source-repository-package` is not related to the `source-repository` field in package descriptions.

I know that these fields are both unrelated and related.

@ulysses4ever thanks for the suggested wording. I incorporated most of it.

Reply:
>> via the same-named parameter if specified
>
> Sorry I don't quite follow @ulysses4ever. Do you mean [PACKAGES] by this?


What I meant is the `-s` parameter that you can read about in `--help`:

```shellsession
❯ cabal get --help
Download/Extract a package's source code (repository).

Usage: cabal get [PACKAGES]

Creates a local copy of a package's source code. By default it gets the source
tarball and unpacks it in a local subdirectory. Alternatively, with -s it will
get the code from the source repository specified by the package.

Flags for get:
...
 -s[[head|this|...]], --source-repository[=[head|this|...]]
                                Copy the package's source repository (ie git
                                clone, darcs get, etc as appropriate).
```

I don't know about any other usage that that field in package descriptions has.

Reply:
Happy to add a mention of `cabal get` while I'm at it.

Reply:
I've done this, added a link to `cabal get`.